### DISCUSSION TITLE
Is there a way to use a custom dictionary to parse the data ?

### DISCUSSION THREAD

**Discussion by shulard** [Enet4/dicom-rs#503] [Q&A] [answered]
Hello !

I discovered this awesome library and would like to use it in a project.

I understand that the default DICOM dictionary is exposed directly. Is there a way to use a customer dictionary when parsing the data? I've almost 10 thousand (most of them are private tags) and I want to be able to parse those data. I'm already using DCMTK so my dictionary is using the same format (I don't code with DCMTK, I use the built applications ^^).

Maybe it's already possible and I haven't found a way to do so… Let me know if you need more information or context…

Thanks !

**Comment by Enet4** [comment, marked as answer]
Thank you for your interest in DICOM-rs!

The current API already features a [`DataDictionary`](https://docs.rs/dicom-core/latest/dicom_core/dictionary/trait.DataDictionary.html) abstraction. It would be possible to build an implementation of this trait containing a list of private tags and then use it when reading and writing DICOM data sets. The simplest implementation would collect them all statically or from a file into map collections. There is no private creator awareness at this level yet, but it may still be possible to take advantage of this trait with a controlled set of private creators. In any case, none of these APIs are set in stone, and it would be great to see some attempts at introducing element dictionaries with private tags.
